src/vcpkg/configuration.cpp

```diff
@@ -53,21 +209,77 @@ namespace
             registries.add_registry(std::move(reg));
         }
 
-        return Configuration{std::move(registries)};
+        Json::Object ce_config_obj;
+        auto maybe_ce_config = r.visit(obj, CeConfigurationDeserializer::instance);
+        if (auto ce_config = maybe_ce_config.get())
+        {
+            ce_config_obj = *ce_config;
+        }
+
+        return Configuration{std::move(registries), ce_config_obj, extra_info};
     }
 
     ConfigurationDeserializer::ConfigurationDeserializer(const Path& configuration_directory)
         : configuration_directory(configuration_directory)
     {
     }
 
+    static Json::Object serialize_configuration_impl(const Configuration& config)
+    {
+        constexpr static StringLiteral REGISTRY_PACKAGES = "packages";
+
+        auto serialize_packages_list = [](vcpkg::View<std::string> packages) {
+            Json::Array ret;
+            for (auto pkg : packages)
+                ret.push_back(Json::Value::string(pkg));
+            return ret;
+        };
+
+        Json::Object obj;
+
+        for (const auto& el : config.extra_info)
+        {
+            obj.insert(el.first.to_string(), el.second);
+        }
+
+        if (config.registry_set.default_registry())
+        {
+            obj.insert(ConfigurationDeserializer::DEFAULT_REGISTRY,
+                       config.registry_set.default_registry()->serialize());
+        }
+
+        auto reg_view = config.registry_set.registries();
+        if (reg_view.size() > 0)
+        {
+            auto& reg_arr = obj.insert(ConfigurationDeserializer::REGISTRIES, Json::Array());
+            for (const auto& reg : reg_view)
+            {
+                auto reg_obj = reg.implementation().serialize();
+                reg_obj.insert(REGISTRY_PACKAGES, serialize_packages_list(reg.packages()));
```

Comment:
```suggestion
                auto& packages = reg_obj.insert(REGISTRY_PACKAGES, Json::Array{});
                for (const auto& pkg : reg.packages())
                    packages.push_back(pkg);
```

something like this instead?